src/index.ts

```diff
@@ -59,7 +59,7 @@ export default function createClient<T>(defaultOptions?: ClientOptions) {
     : { params?: BaseParams };
   type Params<T> = PathParams<T> & MethodParams<T>;
   type RequestBody<T> = T extends { requestBody: any } ? { body: Unwrap<T['requestBody']> } : { body?: never };
-  type FetchOptions<T> = Params<T> & RequestBody<T> & Omit<RequestInit, 'body'>;
+  type FetchOptions<T> = Params<T> & RequestBody<T> & { querySerializer?: (q: any) => string } & Omit<RequestInit, 'body'>;
```

Comment:
OK after all the above convo, one last request for change: naming it to **paramSerializer** rather than querySerializer, for 2 reasons:

- “parameter serialization” is the official term found in the [specification](https://swagger.io/docs/specification/serialization/)
- `paramSerializer` more clearly indicates it modifies the `params` option